android/jni/com/mapswithme/maps/SponsoredHotel.cpp

```diff
@@ -2,70 +2,160 @@
-    {
+  g_framework->RequestBookingMinPrice(hotelId, code, [hotelId](string const & price,
+                                                               string const & currency) {
+    GetPlatform().RunOnGuiThread([=]() {
       JNIEnv * env = jni::GetEnv();
       env->CallStaticVoidMethod(g_hotelClass, g_priceCallback, jni::ToJavaString(env, hotelId),
-                                                               jni::ToJavaString(env, price),
-                                                               jni::ToJavaString(env, currency));
+                                jni::ToJavaString(env, price), jni::ToJavaString(env, currency));
+    });
+  });
+}
+
+// static void nativeRequestInfo(String id, String locale);
+JNIEXPORT void JNICALL Java_com_mapswithme_maps_widget_placepage_SponsoredHotel_nativeRequestInfo(
+    JNIEnv * env, jclass clazz, jstring id, jstring locale)
+{
+  PrepareClassRefs(env, clazz);
+
+  string const hotelId = jni::ToNativeString(env, id);
+  string const code = jni::ToNativeString(env, locale);
+
+  g_framework->RequestBookingInfo(hotelId, code, [hotelId](
+                                                     BookingApi::HotelInfo const & hotelInfo) {
+    GetPlatform().RunOnGuiThread([=]() {
+      JNIEnv * env = jni::GetEnv();
+
+      auto description = jni::ToJavaString(env, hotelInfo.m_description);
+      auto photos =
+          jni::ToJavaArray(env, g_imageClass, hotelInfo.m_photos,
+                           [](JNIEnv * env, BookingApi::HotelPhotoUrls const & item) {
+                             return env->NewObject(g_imageClass, g_imageConstructor,
+                                                   jni::ToJavaString(env, item.m_original),
+                                                   jni::ToJavaString(env, item.m_small));
+                           });
+      auto facilities =
+          jni::ToJavaArray(env, g_facilityTypeClass, hotelInfo.m_facilities,
+                           [](JNIEnv * env, BookingApi::Facility const & item) {
+                             return env->NewObject(g_facilityTypeClass, g_facilityConstructor,
+                                                   jni::ToJavaString(env, item.m_id),
+                                                   jni::ToJavaString(env, item.m_localizedName));
+                           });
+      auto reviews = jni::ToJavaArray(
+          env, g_reviewClass, hotelInfo.m_reviews,
+          [](JNIEnv * env, BookingApi::HotelReview const & item) {
+            return env->NewObject(
+                g_reviewClass, g_reviewConstructor, jni::ToJavaString(env, item.m_reviewNeutral),
+                jni::ToJavaString(env, item.m_reviewPositive),
+                jni::ToJavaString(env, item.m_reviewNegative),
+                jni::ToJavaString(env, item.m_author), jni::ToJavaString(env, item.m_authorPictUrl),
+                item.m_rating, std::chrono::time_point_cast<std::chrono::milliseconds>(item.m_date)
```

Comment:
```#include "std/chrono.hpp"```
Feel free to add whatever usings you need here.
And we do not use the "std::" prefix.